src/v/cluster/tm_stm.cc

```diff
@@ -626,31 +519,51 @@ ss::future<tm_stm::op_status> tm_stm::add_group(
         co_return tm_stm::op_status::unknown;
```

Comment:
Is this "result unknown" or "unknown transaction"? Looks like it is used in both senses

Reply:
i think we should remove the `op_status` completely and rely on `tx::errc`

Reply:
i will do it in follow up pr, it is already a massive change